flatnav/Index.h

```diff
@@ -52,7 +51,7 @@ template <typename dist_t, typename label_t> class Index {
   size_t _node_size_bytes;
   size_t _max_node_count; // Determines size of internal pre-allocated memory
   size_t _cur_num_nodes;
-  std::shared_ptr<DistanceInterface<dist_t>> _distance;
```

Comment:
There is no need for this to be a shared pointer since the index takes ownership of the distance object after it's created!